### PR TITLE
make DeepQNetwork models support paddlepaddle>=1.0.0

### PR DESCRIPTION
**Pull request by zenghsh3**

### CODE REVIEW

File: fluid/DeepQNetwork/DoubleDQN_agent.py [hunk 1]
```diff
@@ -39,41 +39,58 @@ def _get_inputs(self):
...
-            fluid.layers.elementwise_mul(action_onehot, self.pred_value), dim=1)
+            reward = fluid.layers.clip(reward, min=-1.0, max=1.0)
 
-        targetQ_predict_value = self.get_DQN_prediction(next_s, target=True)
+            action_onehot = fluid.layers.one_hot(action, self.action_dim)
+            action_onehot = fluid.layers.cast(action_onehot, dtype='float32')
 
-        next_s_predcit_value = self.get_DQN_prediction(next_s)
-        greedy_action = fluid_argmax(next_s_predcit_value)
+            pred_action_value = fluid.layers.reduce_sum(
```
Comment (TomorrowIsAnOtherDay) [inline fluid/DeepQNetwork/DoubleDQN_agent.py:58]: fluid support `argmax` at the moment.

File: fluid/DeepQNetwork/DoubleDQN_agent.py [hunk 2]
```diff
@@ -39,41 +39,58 @@ def _get_inputs(self):
...
-            dim=1)
-        best_v.stop_gradient = True
+            targetQ_predict_value = self.get_DQN_prediction(next_s, target=True)
 
-        target = reward + (1.0 - fluid.layers.cast(
-            isOver, dtype='float32')) * self.gamma * best_v
-        cost = fluid.layers.square_error_cost(pred_action_value, target)
-        cost = fluid.layers.reduce_mean(cost)
+            next_s_predcit_value = self.get_DQN_prediction(next_s)
+            greedy_action = fluid_argmax(next_s_predcit_value)
```
Comment (TomorrowIsAnOtherDay) [inline fluid/DeepQNetwork/DoubleDQN_agent.py]: we could replace `fluid_argmax`  with `argmax` in fluid.
Please remove fluid_argmax in this folder.

File: fluid/DeepQNetwork/DuelingDQN_agent.py [hunk 3]
```diff
@@ -1,10 +1,10 @@
 #-*- coding: utf-8 -*-
 
+import math
+import numpy as np
 import paddle.fluid as fluid
 from paddle.fluid.param_attr import ParamAttr
-import numpy as np
 from tqdm import tqdm
-import math
 from utils import fluid_flatten
```
Comment (TomorrowIsAnOtherDay) [inline fluid/DeepQNetwork/DuelingDQN_agent.py]: fluid support `flatten`, too.
Please remove file `utils` and use fluid general API.

**Review by TomorrowIsAnOtherDay — CHANGES_REQUESTED** [review]

File: fluid/DeepQNetwork/DoubleDQN_agent.py (hunk 1 above)
Comment (zenghsh3) [reply fluid/DeepQNetwork/DoubleDQN_agent.py:58]: done

File: fluid/DeepQNetwork/DoubleDQN_agent.py (hunk 2 above)
Comment (zenghsh3) [reply fluid/DeepQNetwork/DoubleDQN_agent.py]: done

File: fluid/DeepQNetwork/DuelingDQN_agent.py (hunk 3 above)
Comment (zenghsh3) [reply fluid/DeepQNetwork/DuelingDQN_agent.py]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]

**Pull request merged**